src/main/java/cn/nukkit/player/Player.java

```diff
@@ -1554,6 +1555,12 @@ public boolean onUpdate(int currentTick) {
             this.updateData();
         }
 
+        setFlag(
+                EntityFlag.BLOCKING,
+                this.isSneaking() && this.getInventory().getItemInHand().getId() == ItemIds.SHIELD // Add support to off hand
+        );
+        updateData();
+
         return true;
     }
 
```

Comment:
I think a `isBlocking()` method which basically checks the dataflag would be beneficial for the plugins, can you add it too? :)

Reply:
done :D